src/components/Profile.js

```diff
@@ -121,30 +127,75 @@ function Profile() {
         </AllCollectionsWrapper>
       );
     } else if (showSection === MY_SCHEDULE) {
-      return <div>welcome to my schedule</div>;
+      return <div>welcom to my schedule</div>;
     }
   };
 
+  const getPins = async (id) => {
+    const querySnapshot = await getDocs(collection(db, "user", id, "pin"));
+    let myPins = [];
+    querySnapshot.forEach((doc) => {
+      myPins.push({...doc.data()});
+    });
+    setPins(myPins);
+  };
+
+  useEffect(() => {
+    userData && getPins(userData.id);
+  });
```

Comment:
Why do you use another same useEffect with empty dependency arr?

Reply:
arranged in a useEffect